Makefile

```diff
@@ -1,16 +1,57 @@
 DIST := dist
 FILEBASE := universal-search
+S3_BUCKET_NAME := universal-search
+S3_BUCKET_REGION := us-west-2
+S3_HOSTNAME := https://s3-$(S3_BUCKET_REGION).amazonaws.com
 VERSION := `cat install.rdf | tr '\n' ' ' | sed "s/.*<em:version>\(.*\)<\/em:version>.*/\1/"`
 PWD := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
-XPI := $(PWD)/$(DIST)/$(FILEBASE)-$(VERSION).xpi
+FILENAME := $(FILEBASE)-$(VERSION).xpi
+SIGNED_FILENAME := universal_search_addon-$(VERSION)-fx.xpi
+UPDATE_FILENAME := $(FILEBASE).xpi
+XPI := $(PWD)/$(DIST)/$(FILENAME)
 
-.PHONY: all build
+.PHONY: all build sign tag upload require-amo-env require-aws-env
 
 all:
 	@$(MAKE) build
+	@$(MAKE) sign
+	@$(MAKE) tag
+	@$(MAKE) upload
 
 build:
 	@mkdir -p $(DIST)
-	@rm -rf $(XPI)
-	@zip -r $(XPI) . -x "$(DIST)/*" "Makefile" > /dev/null
-	@echo "Add-on built at $(XPI)"
+	@rm -f $(XPI)
+	@zip -r $(XPI) . -x "$(DIST)/*" ".git/*" "Makefile" > /dev/null
```

Comment:
Needed to add `.git/*` to pass validation; probably should have been there in the first place.